.ci/scripts/test_llama.sh

```diff
@@ -158,7 +158,7 @@ cmake_install_executorch_libraries() {
         -DEXECUTORCH_BUILD_QNN="$QNN" \
         -DQNN_SDK_ROOT="$QNN_SDK_ROOT" \
         -Bcmake-out .
-    cmake --build cmake-out -j9 --target install --config "$CMAKE_BUILD_TYPE"
+    cmake --build cmake-out -j$(($(nproc) - 1)) --target install --config "$CMAKE_BUILD_TYPE"
```

Comment:
This script is also used to test locally on macOS, so we cannot just use nproc. 


Overall what I would suggest is abstracting this out logic out into a common script and perhaps sharing the logic. For example, create `tools/common_build.sh`

```bash
core_count() {
  if [[ "$(uname)" == "Darwin" ]]; then
    num_cores=$(sysctl -n hw.ncpu)
  else
    num_cores=$(nproc)
  fi
  echo $((num_cores - 1))
}
```